src/reconnector.js

```diff
@@ -39,19 +39,24 @@ class Reconnector {
     });
 
     this.backoff.on('fail', () => {
-      this.client.logger.error('Failed to reconnect to the streaming service. Attempting to connect with new channel.');
-      // attempt with a new channel
-      this.cleanupReconnect();
-      this.client.connect();
+      this.hardReconnect();
     });
 
     // self bound methods so we can clean up the handlers
     this._cleanupReconnect = this.cleanupReconnect.bind(this);
-    this.client.on('connected', this._cleanupReconnect);
+    this.client.on('connected', () => {
+      this._hasConnected = true;
+      this._cleanupReconnect();
+    });
 
     // disable reconnecting when there's an auth failure
     this.client.on('sasl:failure', (err) => {
       if (!err || err.condition !== 'temporary-auth-failure') {
+        // an auth failure when it's already connected once means channel expiry
+        if (err && err.condition === 'not-authorized' && this._hasConnected) {
```

Comment:
NIT:  Pull out this logic into a variable called `channelExpired` or something like that and you won't need the comment